### PR TITLE
Adds AMAZON.LITERAL built-in type

### PR DESCRIPTION
**Pull request by cnadeau**
Fixes issue virtual-alexa #66 Adds AMAZON.LITERAL built-in type

https://github.com/bespoken/virtual-alexa/issues/66

### CODE REVIEW

File: src/SampleUtterances.ts [hunk 1]
```diff
@@ -93,8 +93,22 @@ export class SamplePhrase {
         const startIndex = phrase.indexOf("{");
         if (startIndex !== -1) {
             const endIndex = phrase.indexOf("}", startIndex);
-            this.slotNames.push(phrase.substring(startIndex + 1, endIndex));
-            phrase = phrase.substring(0, startIndex).trim() + "(.*)" + phrase.substring(endIndex + 1).trim();
+            const slotName = phrase.substring(startIndex + 1, endIndex);
+
```
Comment (jkelvie) [inline src/SampleUtterances.ts]: Please remove extra space here

File: src/SampleUtterances.ts [hunk 2]
```diff
@@ -93,8 +93,22 @@ export class SamplePhrase {
...
+            const pipe = slotName.indexOf("|");
+
+            if(pipe === -1){
+                this.slotNames.push(slotName);
+                phrase = phrase.substring(0, startIndex).trim() + "(.*)" + phrase.substring(endIndex + 1).trim();
+            } else{
```
Comment (jkelvie) [inline src/SampleUtterances.ts]: This should be automatically linted, but there should be a space after else

File: src/SampleUtterances.ts [hunk 3]
```diff
@@ -93,8 +93,22 @@ export class SamplePhrase {
...
+                // Literal are in the format "sample { <literal sample> | <slotname>}"
+                // e.g.: "I'm an {aquarius | literal}"
+                const literalSample = slotName.substring(0, pipe);
+                const literalSlotName = slotName.substring(pipe + 2, slotName.length);
+                this.slotNames.push(literalSlotName);
+                phrase = phrase.substring(0, startIndex).trim() + literalSample.trim() + phrase.substring(endIndex + 1).trim();
```
Comment (jkelvie) [inline src/SampleUtterances.ts]: This is going to make the phrase match ONLY if the example literal value is in the utterance - I believe it is better not to do this (i.e., it will only match if the utterance is "second literal input"). On the one hand, this will improve the performance when the suggested phrases are used. But our logic is that we will then restrict it to ONLY those enumerated phrases, so no other phrases will work.

Our matching algorithm is based on looking at how precisely the text of the phrase as well as the values of the slots are matched. It is simplistic compared to a real NLU. But it should perform better, and for a greater variety of utterances, if this is left out.

So long story short, this line should be taken out, and the phrase on line 102 should be moved outside the if condition (as it will be parsed the same way for both).

**Comment by cnadeau** [comment]
You're right, coded this way, it's more like a custom type with **very** specific values. But, as we can see, a single literal without anything around a sample will lead to anything being matched, which is a behavior we really want to avoid, no?
